src/FsCheck/Gen.fs

```diff
@@ -168,6 +168,18 @@ module Gen =
     let elements xs = 
         choose (0, (Seq.length xs)-1) |> map (flip Seq.nth xs)
 
+    [<CompiledName("GrowingElements")>]
+    ///Build a generator that takes a list of elements of increasing size and
+    ///chooses among an initial segment of the list. The size of this initial
+    ///segment increases with the size parameter.
+    //[category: Creating generators]
+    let growingElements xs =
+        let l = Seq.length xs
```

Comment:
Can you make sure that the xs sequence gets enumerated only once? This is a reasonable and intuitive expectation I'd like to uphold. 
In this case I would suggest calling `Seq.toArray` into a variable outside the closure below and then using `arr.Count` and `arr` instead of `xs` inside the generator.